### PR TITLE
elementary: rename Activew_View to Spotlight.

### PR DESCRIPTION
**Pull request by Bluebugs**
View is something that is expected in the context of MVVM, so using it somewhere else is
going to lead to some confusion. Spotlight does descrive the objective of all of this
widget in actually a more explicit way as they all give the spotlight to one sub widget
at a time.

I have also renamed the View_Manager to be just Manager as the View there wasn't useful.

### CODE REVIEW

**Comment by segfaultxavi** [comment]
**More context for the next reviewer:** Yes. We are all sick and tired of renaming this widget. But having `View` used for anything other than MVVM is going to be confusing no matter what. Hence this third and **last** rename.

File: src/bin/elementary/test.c [hunk 1]
```diff
@@ -1117,9 +1117,9 @@ my_win_main(const char *autorun, Eina_Bool test_win_only)
 
    //------------------------------//
    ADD_TEST_EO(NULL, "Tab Pager", "Efl.Ui.Tab_Pager", test_ui_tab_pager);
-   ADD_TEST_EO(NULL, "Active View", "Efl.Ui.Active_View Plain", test_ui_active_view_plain);
-   ADD_TEST_EO(NULL, "Active View", "Efl.Ui.Active_View Scroll", test_ui_active_view_scroll);
-   ADD_TEST_EO(NULL, "Active View", "Efl.Ui.Active_View Stack", test_ui_active_view_stack);
+   ADD_TEST_EO(NULL, "Active View", "Efl.Ui.Spotlight Plain", test_ui_spotlight_plain);
```
Comment (segfaultxavi) [inline src/bin/elementary/test.c]: Better rename the group too, right?

File: src/lib/elementary/Efl_Ui.h [hunk 2]
```diff
@@ -227,18 +227,18 @@ EAPI void efl_ui_focus_relation_free(Efl_Ui_Focus_Relations *rel);
...
-typedef Eo Efl_Ui_Active_View_Indicator;
-
-# include <efl_ui_active_view_container.eo.h>
-# include <efl_ui_active_view_view_manager.eo.h>
-# include <efl_ui_active_view_indicator.eo.h>
-# include <efl_ui_active_view_indicator_icon.eo.h>
-# include <efl_ui_active_view_view_manager_scroll.eo.h>
-# include <efl_ui_active_view_view_manager_stack.eo.h>
-# include <efl_ui_active_view_util.eo.h>
+#define _EFL_UI_SPOTLIGHT_MANAGEREO_CLASS_TYPE
```
Comment (segfaultxavi) [inline src/lib/elementary/Efl_Ui.h]:
```suggestion
#define _EFL_UI_SPOTLIGHT_MANAGER_EO_CLASS_TYPE
```

File: src/lib/elementary/efl_ui_spotlight_container.eo [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+   [[Information regarding transition events.]]
+   from : int; [[The index from where the transition started, -1 if not known.]]
+   to : int; [[The index to where the transition is headed, -1 if not known.]]
+}
+
+class @beta Efl.Ui.Spotlight.Container extends Efl.Ui.Layout_Base implements Efl.Pack_Linear
+{
+   [[The Spotlight widget is a container for other sub-widgets, where only one sub-widget is active at any given time.
+
+     Sub-widgets are added using the @Efl.Pack_Linear interface and the spotlight is selected using @.active_index.
```
Comment (segfaultxavi) [inline src/lib/elementary/efl_ui_spotlight_container.eo]:
```suggestion
     Sub-widgets are added using the @Efl.Pack_Linear interface and the active one (the one in the "spotlight") is selected using @.active_index.
```

File: src/lib/elementary/efl_ui_spotlight_container.eo [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        [[An indicator object to use, which will display the current state of the spotlight (number of sub-widgets and active one).
+          When this object is set, it is immediately updated to reflect the current state of the widget.
+          Its location inside the container is controlled by the @.spotlight_manager.
+        ]]
+        values {
+          indicator : Efl.Ui.Spotlight.Indicator @owned; [[The Indicator object or $NULL.]]
+        }
+      }
+      @property active_index {
+         [[Currently active sub-widget among all the sub-widgets added to this widget.
```
Comment (segfaultxavi) [inline src/lib/elementary/efl_ui_spotlight_container.eo]:
```suggestion
         [[Currently active sub-widget (the one with the spotlight) among all the sub-widgets added to this widget.
```

File: src/lib/elementary/efl_ui_spotlight_container.eo [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+           An animation might be triggered to give the new sub-widget the spotlight and come into position.
+         ]]
+         params {
+            widget : Efl.Gfx.Entity; [[Sub-Widget to add and set to be the spotlight sub-widget.]]
+         }
+      }
+      pop @beta {
+         [[Removes the sub-widget that has the spotlight from the widget.
+
+           The sub-widgets behind it naturally flow down so the next one get the spotlight. This is the same behavior as a pop operation on a stack.
```
Comment (segfaultxavi) [inline src/lib/elementary/efl_ui_spotlight_container.eo]:
```suggestion
           The sub-widgets behind it naturally flow down so the next one gets the spotlight. This is the same behavior as a pop operation on a stack.
```

File: src/lib/elementary/efl_ui_spotlight_indicator.eo [hunk 6]
```diff
@@ -1,25 +1,25 @@
-abstract @beta Efl.Ui.Active_View.Indicator extends Efl.Object {
-  [[Object used by @Efl.Ui.Active_View.Container to render an indication of the active view's position among
+abstract @beta Efl.Ui.Spotlight.Indicator extends Efl.Object {
+  [[Object used by @Efl.Ui.Spotlight.Container to render an indication of the active view's position among
```
Comment (segfaultxavi) [inline src/lib/elementary/efl_ui_spotlight_indicator.eo]:
```
  [[Object used by @Efl.Ui.Spotlight.Container to render an indication of the active widgets's position among
      the rest of the container's widgets.
```

**Comment by zmike** [comment]
I can't tell if I hate it or love it!

**Comment by segfaultxavi** [comment]
I'm THAT good at naming things

**Comment by segfaultxavi** [comment]
Everything's fine by me. When @zmike decides whether he loves or hates this, we merge.

**Pull request merged**